odoo/models.py

```diff
@@ -2413,6 +2418,24 @@ def _read_group_format_result(self, data, annotated_groupbys, groupby, domain):
         del data['id']
         return data
 
+    @api.model
+    def _read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True):
+        """
+        Executes exactly what the public read_group, except it doesn't order the manyToOne on their names but on their ID instead
+        """
+        if not orderby:
+            if isinstance(groupby, str):
+                groupby = [groupby]
+            groupby_list = groupby[:1] if lazy else groupby
+            order_list = []
+            for order_spec in groupby_list:
+                field_name = order_spec.split(":")[0]  # field name could be formatted like "field:group_func"
+                if self._fields[field_name].type == 'many2one':
+                    order_spec = f"{field_name}.id"  # do not order by comodel's order
+                order_list.append(order_spec)
+            order = ','.join(g for g in order_list)
```

Comment:
The method `join` takes an iterable as parameter, and `order_list` is an iterable.  Why creating a generator that iterates over `order_list` for not doing anything special?
```suggestion
            order = ','.join(order_list)
```